lib/cracker.rb

```diff
@@ -11,7 +11,10 @@ def reverse_shifts(ciphertext, known_end = " end")
   end
 
   def root_codes(shifts, offsets)
-    shifts.zip(offsets).map {|shift, offset| shift - offset.to_i}
+    shifts.zip(offsets).map do |shift, offset|
+      diff = shift - offset.to_i
+      diff > 0 ? diff : full_shift(diff)
```

Comment:
found this bug after adding the _very last_ extra test

`Testing: It's Worth It™`